packages/eas-cli/src/commandUtils/EasCommand.ts

```diff
@@ -27,7 +28,9 @@ export default abstract class EasCommand extends Command {
   // eslint-disable-next-line async-protect/async-suffix
   async run(): Promise<any> {
     await initAnalyticsAsync();
-    await this.applyCliConfigAsync();
+    if (await pkgDir()) {
+      await this.applyCliConfigAsync();
+    }
```

Comment:
I'd prefer to re-write `applyCliConfigAsync` so it checks whether `eas.json` exists in the current directory or in any of the parent directories.
@wkozyra95 what do you think?

Reply:
I agree. Reading eas.json already is written in a way that should not fail, I think the only part of that function that needs to modified is `findProjectRootAsync`

Also I think that commands that can be ran outside the project dir should be explicitly marked, e.g. the same as `requiresAuthentication` flag

Reply:
> I'd prefer to re-write applyCliConfigAsync... the only part of that function that needs to modified is findProjectRootAsync

If we are not inside of a project, why would we want `applyCliConfigAsync` to run? `applyCliConfigAsync` seems only relevant for commands that may want to use version control. If you are not inside of a project, there is no version control and thus why would we run `applyCliConfigAsync`?

Reply:
> I'd prefer to re-write applyCliConfigAsync... the only part of that function that needs to modified is findProjectRootAsync

If we are not inside of a project, why would we want `applyCliConfigAsync` to run? `applyCliConfigAsync` seems only relevant for commands that may want to use version control. If you are not inside of a project, there is no version control and thus why would we run `applyCliConfigAsync`?

Reply:
currently, you are duplicating how to find project root, here in applyCliConfigAsync

Reply:
Agreed with Wojtek. Also, we don't need to keep the current name (`applyCliConfigAsync`). I assume this is what you don't like. We can change it so it tell the reader what it does :)

Reply:
OK, I applied this suggestion: `commands that can be ran outside the project dir should be explicitly marked, e.g. the same as requiresAuthentication flag`

* This avoid us duplicating the call to `pkgDir`.
* This command works as written as long as  eas.json exists in the current directory or in any of the parent directories.`

Is this closer to what you guys are thinking? If not, maybe I can zoom briefly tomorrow morning to understand a bit better.

Reply:
I was thinking about changing that code that applyCliConfigAsync is always run, but in the case of `mustBeRunInsideAnExpoProject=false` it won't throw an error.

For the current set of commands that do not require project your code and what I described is equivalent because those commands do not use anything that applyCliConfigAsync my do. It might be different in the future.